sdk/batch/azure-compute-batch/src/main/java/com/azure/compute/batch/BatchAsyncClient.java

```diff
@@ -10809,14 +12315,8 @@ public Mono<BatchApplication> getApplication(
      * @return contains information about an application in an Azure Batch Account on successful completion of {@link
      *     Mono}.
      */
-    @Generated
-    @ServiceMethod(returns = ReturnType.SINGLE)
-    public Mono<BatchApplication> getApplication(String applicationId) {
-        // Generated convenience method for getApplicationWithResponse
-        RequestOptions requestOptions = new RequestOptions();
-        return getApplicationWithResponse(applicationId, requestOptions)
-                .flatMap(FluxUtil::toMono)
-                .map(protocolMethodData -> protocolMethodData.toObject(BatchApplication.class));
+    public Mono<BatchApplication> getApplication(String applicationId, BatchBaseOptions options) {
```

Comment:
I wonder if we should just go ahead and subclass every one of these sets of options. If we reference one of the common classes directly, then if we need to add an option that's specific to an operation, it's a breaking change.

Example:

Imagine we ship this code as-is, then later we decide getApplication() should have an "expand" query parameter. Now we need to change the method signature from `getApplication(String, BatchBaseOptions)` to `getApplication(String, GetBatchApplicationOptions)`.

Alternatively, if we just make an empty subclass of BatchBaseOptions right now, the signature will be `getApplication(String, GetBatchApplicationOptions)` in both cases, but GetBatchApplicationOptions will have simply gained an optional property.